File: bintree_py/test/testavl01.py
from .testbase import print_tree
from binarytree import AvlTree


def do_task():
    avl = AvlTree()

    for value in [10, 20, 30, 40, 50, 25]:
        avl.insert(value)

    print('\n count:', avl.count())
    print('\n min:', avl.min())
    print('\n max:', avl.max())
    print('\n contain:', avl.contain(50))
    print('\n height:', avl.height())

    print('\n print tree')
    print_tree(avl)

    print()

File: bintree_java/src/test/TestAvl01.java
package test;


import binarytree.AvlTree;



public class TestAvl01 extends TestBase {

    @Override
    public void doTask() {
        var avl = new AvlTree<Integer>();

        for ( var value : new int[] {10, 20, 30, 40, 50, 25} )
            avl.insert(value);

        System.out.println("\n count: " + avl.count());
        System.out.println("\n min: " + avl.min());
        System.out.println("\n max: " + avl.max());
        System.out.println("\n contain: " + avl.contain(50));
        System.out.println("\n height: " + avl.height());

        System.out.println("\n print tree: ");
        printTree(avl);

        System.out.println();
    }

}
